doc/guide.md

```diff
@@ -0,0 +1,263 @@
+# User Guide
+
+## Background
+
+The goal of this library is to automatically calculate an optimal point that is always guaranteed to fall inside a
+polygon. It was conceived when trying to optimize label placement in geometric shape data. Initial attempts at label
+placement used the centroid. However, non-convex shapes were quite common and often labels would get drawn outside
+the shape. Additionally, many shapes had centroids that resulted in label locations that did not look good to the naked
+eye, typically residing in small areas of the shape such as narrow strips or near holes.
+
+After some visual analysis and research, it became obvious that the optimal location for label placement would be at
+the center of the largest circle that can be inscribed in the shape. This location is optimal as it is the part of the
+shape that has the most space around it.
+
+![Motivation](images/motivation.png)
+
+This circle, known as the maximum inscribed circle, will always have its center lie on the
+[medial axis](https://en.wikipedia.org/wiki/Medial_axis) of the shape. The medial axis can be thought of as the bone
+structure of a shape, where every point on the medial axis has at least two equidistant closest points on the shape.
+Note, the following example is a manual approximation and is not 100% accurate, particularly in the curved segments.
```

Comment:
I'd put this after the medial axis image, in italics, so it feels more like a footnote associated with the image.

Reply:
Good point. I like that.